python-package/setup.py

```diff
@@ -292,6 +292,13 @@ def run(self):
               'scipy',
               'scikit-learn!=0.22.0'
           ],
+          extras_requires={
+              'dask': [
+                'dask>=0.16.0',
+                'distributed>=1.15.2'
+
+              ],
+          },
```

Comment:
I agree with this, making `dask` an optional dependency that people install separately or with `pip install lightgbm[dask]`. I definitely don't want to force all existing `lightgbm` users to install `dask`.

I think this section should also include `pandas`  though...`pandas` is not strictly required for `lightgbm`: https://github.com/microsoft/LightGBM/blob/cf69591b68c68a78ea4064e6941b8270b8a00eef/python-package/lightgbm/compat.py#L61

Reply:
I am not sure if adding pandas as package dependency is necessary, since the dask integration does not need the pandas. 

The only place where it is required are tests but I was not able to find where are listed dev dependencies. Can you please point me to correct place where I can add the dev dependencies?

Reply:
oh sure!

If it's a test-only dependency, then you're already covered. We install `pandas` into our testing conda environment:

* https://github.com/microsoft/LightGBM/blob/cf69591b68c68a78ea4064e6941b8270b8a00eef/.ci/test.sh#L73
* https://github.com/microsoft/LightGBM/blob/cf69591b68c68a78ea4064e6941b8270b8a00eef/.ci/test_windows.ps1#L25